lib/utils/helpers.coffee

```diff
@@ -0,0 +1,134 @@
+###
+Copyright 2016-2017 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+###
+
+Promise = require('bluebird')
+
+exports.getGroupDefaults = (group) ->
+	_ = require('lodash')
+
+	return _.chain(group)
+		.get('options')
+		.map (question) ->
+			return [ question.name, question.default ]
+		.fromPairs()
+		.value()
+
+exports.stateToString = (state) ->
+	_str = require('underscore.string')
+	chalk = require('chalk')
+
+	percentage = _str.lpad(state.percentage, 3, '0') + '%'
+	result = "#{chalk.blue(percentage)} #{chalk.cyan(state.operation.command)}"
+
+	switch state.operation.command
+		when 'copy'
+			return "#{result} #{state.operation.from.path} -> #{state.operation.to.path}"
+		when 'replace'
+			return "#{result} #{state.operation.file.path}, #{state.operation.copy} -> #{state.operation.replace}"
+		when 'run-script'
+			return "#{result} #{state.operation.script}"
+		else
+			throw new Error("Unsupported operation: #{state.operation.type}")
+
+exports.sudo = (command) ->
+	_ = require('lodash')
+	os = require('os')
+
+	if os.platform() isnt 'win32'
+		console.log('If asked please type your computer password to continue')
+
+	command = _.union(_.take(process.argv, 2), command)
+	presidentExecuteAsync = Promise.promisify(require('president').execute)
+	return presidentExecuteAsync(command)
+
+exports.getManifest = (image, deviceType) ->
+	rindle = require('rindle')
+	imagefs = require('resin-image-fs')
+	resin = require('resin-sdk-preconfigured')
+
+	# Attempt to read manifest from the first
+	# partition, but fallback to the API if
+	# we encounter any errors along the way.
+	imagefs.read
+		image: image
+		partition: 1
+		path: '/device-type.json'
+	.then(rindle.extractAsync)
```

Comment:
@pimterry There is a `readFile` method that simply returns a `Promise` with a string value containing the contents. However, when I try to call this method, I get TypeScript errors saying it does not exist. Where are the TypeScript definitions for `resin-image-fs`?

Reply:
That's because resin-image-fs itself doesn't have any types, it's just got the adhoc ones that are in this codebase, which only covers the stuff we're already using, and doesn't know anything about different versions of resin-image-fs. The file you're looking for is here: https://github.com/resin-io/resin-cli/blob/master/typings/resin-image-fs.d.ts

Don't know how familiar with TS you are, let me know you need a hand.